src/components/views/CheckoutPage.ts

```diff
@@ -0,0 +1,366 @@
+        this.container.classList.remove('popup_active');
+        document.body.classList.remove('inactive-order');
+        document.body.removeEventListener('click', (e: MouseEvent) => {
+            this.checkClickWhenOrderOpen(e);
+        });
+    }
+
+    private checkClickWhenOrderOpen(e: MouseEvent) {
+        const target = e.target as HTMLElement;
+        const currentTarget = e.currentTarget as HTMLElement;
+        if (target.classList.contains('inactive-order') && !currentTarget.classList.contains('order-container')) {
+            this.hideModal();
+        }
+    }
+
+    private checkIsUserAgree() {
+        if (this.contactAgree.checked) {
+            this.agree = true;
+            return true;
+        }
+        this.agree = false;
+        return false;
+    }
+
+    private checkFirstName(firstname: string) {
+        this.firstname = firstname;
+    }
+
+    private checkLastName(lastname: string) {
+        this.lastname = lastname;
+    }
+
+    private checkAddress(address: string) {
+        this.address = address;
+    }
+
+    private checkPhone(contactPhone: string) {
+        this.phone = contactPhone;
+    }
+
+    private checkMail(contactMail: string) {
+        this.mail = contactMail;
+    }
+
+    private checkCardNumber(cardNumber: string) {
+        this.cardNumber = cardNumber;
+        const paymentSystemIdStr = cardNumber.slice(0, 1);
+        const paymentSystemId = Number(paymentSystemIdStr);
+        switch (paymentSystemId) {
+            case 4:
+                return 'visa-logo';
+            case 5:
+                return 'mastercard-logo';
+            case 3:
+                return 'a-express-logo';
+            default:
+                return 'nocard-logo';
+        }
+    }
+
+    private checkCardValid(paymentInputValid: HTMLInputElement) {
+        if (!paymentInputValid.value.match(/[0-9]/g)) {
+            paymentInputValid.value = '';
+            return;
+        }
+        if (paymentInputValid.value.length === 3) {
+            if (paymentInputValid.value[2] === '/') {
+                return;
+            } else {
+                paymentInputValid.value = paymentInputValid.value
+                    .split('')
+                    .map((item, index) => {
+                        if (index === 2) {
+                            return (item = `/${item}`);
+                        } else {
+                            return item;
+                        }
+                    })
+                    .join('');
+            }
+        }
+        if (paymentInputValid.value.length > 5) {
+            paymentInputValid.value = paymentInputValid.value.substring(0, 5);
+            this.cardValidTime = paymentInputValid.value;
+        }
+    }
+
+    private checkCardCVV(paymentInputCvv: string) {
+        this.cvv = paymentInputCvv;
+    }
+
+    private checkAllInputsFull() {
+        if (
+            this.firstname.length !== 0 &&
+            this.lastname.length !== 0 &&
```

Comment:
Looks like you could put all these ....length into method and do not  duplicate them.

Reply:
fix it